pkg/util/util.go

```diff
@@ -1456,217 +1452,120 @@ func VerifyExternalSecretsDeletion(kms nbv1.KeyManagementServiceSpec, namespace
+		config[vaultClientKey] = clientKeyFileAddr
 	}
-	return tlsConfig, nil
+	return nil
 }
 
 // PutSecret writes the secret to the secrets store
-func PutSecret(client *vaultApi.Client, secretName, secretValue, secretPath string, backendPath string) error {
+func PutSecret(client secrets.Secrets, secretName, secretValue, secretPath string, backendPath string) error {
 
-	// Build Secret
+	keyContext := map[string]string{}
 	data := make(map[string]interface{})
-	v2, err := isKV2(client, backendPath)
-	if err != nil {
-		return err
-	}
-	if v2 {
-		data["data"] = map[string]interface{}{secretName: secretValue}
-	} else {
-		data[secretName] = secretValue
-	}
+	data[secretName] = secretValue
 
-	_, err = client.Logical().Write(secretPath, data)
+	err := client.PutSecret(secretPath, data, keyContext)
 	if err != nil {
+		log.Errorf("KMS PutSecret: secret path %v value %v, error %v", secretPath, secretValue, err)
 		return err
 	}
+
 	return nil
 }
 
 // GetSecret reads the secret to the secrets store
-func GetSecret(client *vaultApi.Client, secretName, secretPath string, backendPath string) (string, error) {
-	secret, err := client.Logical().Read(secretPath)
+func GetSecret(client secrets.Secrets, secretName, secretPath string, backendPath string) (string, error) {
+	keyContext := map[string]string{}
+	s, err := client.GetSecret(secretPath, keyContext)
 	if err != nil {
+		log.Errorf("KMS GetSecret: secret path %v, error %v", secretPath, err)
 		return "", err
 	}
-	if secret == nil {
-		return "", nil
-	}
-	v2, err := isKV2(client, backendPath)
-	if err != nil {
-		return "", err
-	}
-	if v2 {
-		rootKey, ok := secret.Data["data"].(map[string]interface{})
-		// data property does not exist in secret (in kv2 a secret may exist but have deletion timestamp)
-		if !ok {
-			return "", nil
-		}
-		rootKeyData := rootKey[secretName]
-		if rootKeyData != nil {
-			if rootKeyStr, ok := rootKeyData.(string); ok {
-				return rootKeyStr, nil
-			}
-		}
-	} else {
-		rootKey := secret.Data[secretName]
-		if rootKey != nil {
-			if rootKeyStr, ok := rootKey.(string); ok {
-				return rootKeyStr, nil
-			}
-		}
-	}
-	log.Infof("get secret: found other secrets in external KMS path, but not path: %v, secret name: %v", backendPath, backendPath)
-	return "", nil
+
+	return s[secretName].(string), nil
 }
 
 // DeleteSecret deletes the secret from the secrets store
-func DeleteSecret(client *vaultApi.Client, secretPath string) error {
-	_, err := client.Logical().Delete(secretPath)
+func DeleteSecret(client secrets.Secrets, secretPath string) error {
+	keyContext := map[string]string{}
+	err := client.DeleteSecret(secretPath, keyContext)
 	if err != nil {
-		log.Infof("DeleteSecret: err %+v", err)
+		log.Errorf("KMS DeleteSecret: secret path %v, error %v", secretPath, err)
 		return err
 	}
-	log.Infof("DeleteSecret: deletion from secret path %+v succeded", secretPath)
-
 	return nil
 }
 
-func isKV2(client *vaultApi.Client, backend string) (bool, error) {
```

Comment:
Where are you detecting the version now?

Reply:
Should we worry about the vault API version? No longer working with vaultApi.Client,  hope libopenstorage/secrets would hide those details.